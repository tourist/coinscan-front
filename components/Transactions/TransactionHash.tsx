import Box from '@mui/material/Box';

import ChainScannerLink from '../Addresses/ChainScannerLink';
import CopyToClipboard from '../CopyToClipboard';
import Hash from '../Addresses/Hash';
import settings from '../../settings.json';

type TransactionHashProps = { txn: string; short?: boolean };

const TransactionHash = ({ txn, short }: TransactionHashProps) => {
  const link = `${settings.scannerTxnLink}${txn}`;
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        a: { display: 'flex' },
      }}
    >
      <Hash short={short} text={txn} />
      <CopyToClipboard text={txn} />
      <Box sx={{ ml: 1 }} />
      <ChainScannerLink title={link} link={link} />
    </Box>
  );
};

export default TransactionHash;
