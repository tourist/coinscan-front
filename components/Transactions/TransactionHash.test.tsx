import { screen, render } from '@testing-library/react';
import snapshotDiff from 'snapshot-diff';
import TransactionHash from './TransactionHash';

test('render transactions hash', async () => {
  const { container } = render(
    <TransactionHash txn="0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d" />
  );
  screen.getByText(
    '0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d'
  );
  expect(
    screen.getAllByTitle(
      'https://polygonscan.com/tx/0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d'
    )[0]
  ).toHaveAttribute(
    'href',
    'https://polygonscan.com/tx/0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d'
  );
  expect(container).toMatchSnapshot();
});

test('render transactions hash with short prop', async () => {
  const { container } = render(
    <TransactionHash txn="0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d" />
  );
  screen.getByText(
    '0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d'
  );
  expect(container).toMatchSnapshot();
});

test('render wallet with short prop set', async () => {
  const { asFragment, rerender } = render(
    <TransactionHash txn="0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d" />
  );
  const firstRender = asFragment();
  rerender(
    <TransactionHash
      txn="0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d"
      short
    />
  );
  screen.getByText(
    '0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d'
  );
  expect(snapshotDiff(firstRender, asFragment())).toMatchSnapshot(
    'rerender with short prop'
  );
});
