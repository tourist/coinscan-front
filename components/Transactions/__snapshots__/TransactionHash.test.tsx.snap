// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render transactions hash 1`] = `
<div>
  <div
    class="MuiBox-root css-1gvmo7"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-rdc7yt-MuiTypography-root"
      title="0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d"
    >
      0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d
    </p>
    <svg
      aria-hidden="true"
      aria-label="copy to clipboard"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-eibj8z-MuiSvgIcon-root"
      data-mui-internal-clone-element="true"
      data-testid="ContentCopyIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
      />
    </svg>
    <div
      class="MuiBox-root css-1isemmb"
    />
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1dkk3ib-MuiTypography-root-MuiLink-root"
      href="https://polygonscan.com/tx/0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d"
      rel="noopener noreferrer"
      target="_blank"
      title="https://polygonscan.com/tx/0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-eibj8z-MuiSvgIcon-root"
        data-testid="OpenInNewIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
  </div>
</div>
`;

exports[`render transactions hash with short prop 1`] = `
<div>
  <div
    class="MuiBox-root css-1gvmo7"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-rdc7yt-MuiTypography-root"
      title="0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d"
    >
      0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d
    </p>
    <svg
      aria-hidden="true"
      aria-label="copy to clipboard"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-eibj8z-MuiSvgIcon-root"
      data-mui-internal-clone-element="true"
      data-testid="ContentCopyIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
      />
    </svg>
    <div
      class="MuiBox-root css-1isemmb"
    />
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1dkk3ib-MuiTypography-root-MuiLink-root"
      href="https://polygonscan.com/tx/0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d"
      rel="noopener noreferrer"
      target="_blank"
      title="https://polygonscan.com/tx/0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-eibj8z-MuiSvgIcon-root"
        data-testid="OpenInNewIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
  </div>
</div>
`;

exports[`render wallet with short prop set: rerender with short prop 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,11 +1,11 @@
  <DocumentFragment>
    <div
      class=\\"MuiBox-root css-1gvmo7\\"
    >
      <p
-       class=\\"MuiTypography-root MuiTypography-body2 css-rdc7yt-MuiTypography-root\\"
+       class=\\"MuiTypography-root MuiTypography-body2 css-18mpm0x-MuiTypography-root\\"
        title=\\"0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d\\"
      >
        0x6399e5841ee8b11186186c1a3aed92d93041e22bef09f599460735492b57851d
      </p>
      <svg"
`;
